src/_pytest/_io/saferepr.py

```diff
@@ -54,9 +54,13 @@ def __init__(self, maxsize: Optional[int]) -> None:
         # truncation.
         self.maxstring = maxsize if maxsize is not None else 1_000_000_000
         self.maxsize = maxsize
+        self.use_ascii = use_ascii
 
     def repr(self, x: object) -> str:
         try:
+            if self.use_ascii:
+                return ascii(x)
+
             s = super().repr(x)
```

Comment:
```suggestion
            if self.use_ascii:
                s = ascii(x)
            else:
                s = super().repr(x)
```

We don't want `use_ascii=True` to break the length limit!

Reply:
Ah, I see you commited it yourself! Thank you!